crates/iceberg/src/expr/mod.rs

```diff
@@ -0,0 +1,49 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! This module contains expressions used in apache iceberg.
+
+mod bound;
+pub use bound::*;
+mod unbound;
+pub use unbound::*;
+
+/// Operators used in expressions.
+#[allow(missing_docs)]
+pub enum Operator {
+    IsNull,
+    NotNull,
+    IsNan,
+    NotNan,
+    LessThan,
+    LessThanOrEq,
+    GreaterThan,
+    GreaterThanOrEq,
+    Eq,
+    NotEq,
+    In,
+    NotIn,
+    Not,
+    And,
+    Or,
+    StartsWith,
+    NotStartsWith,
+    Count,
+    CountStar,
+    Max,
+    Min,
```

Comment:
I see java implementation has aggregations: https://github.com/apache/iceberg/blob/8271791061e72dbf554028e477746e89fca9ec02/api/src/main/java/org/apache/iceberg/expressions/Expression.java#L47

I think this is meaningful since we can do optimization with aggregations. For example, `select max(a) for t` we don't need to touch data file if there are min/max index for column a in manifest file.

Reply:
Those aggregations are tricky, since we need to if the min/max aren't truncated. I also see value in `count(passengers) > 2`

Reply:
Sounds reasonable to me. We can remove this first.